da/mock/mock.go

```diff
@@ -71,6 +73,9 @@ func (m *MockDataAvailabilityLayerClient) SubmitBlock(block *types.Block) da.Res
 func (m *MockDataAvailabilityLayerClient) CheckBlockAvailability(header *types.Header) da.ResultCheckBlock {
 	hash := header.Hash()
 	_, err := m.dalcKV.Get(hash[:])
+	if errors.Is(err, badger.ErrKeyNotFound) {
+		return da.ResultCheckBlock{DAResult: da.DAResult{Code: da.StatusSuccess}, DataAvailable: false}
```

Comment:
This is OK - block not found when checking block availability is a normal situation.